vendor-extra/LiberoContentBundle/tests/EventListener/BuildView/FrontTitleTeaserListenerTest.php

```diff
@@ -0,0 +1,143 @@
+        $this->assertSame('@LiberoPatterns/teaser.html.twig', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    public function nodeProvider() : iterable
+    {
+        yield 'different namespace' => ['<front xmlns="http://example.com">foo</front>'];
+        yield 'different element' => ['<foo xmlns="http://libero.pub">foo</foo>'];
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_is_not_the_teaser_template() : void
+    {
+        $listener = new FrontTitleTeaserListener($this->createFailingConverter());
+
+        $element = $this->loadElement('<front xmlns="http://libero.pub"><title>foo</title></front>');
+
+        $event = new BuildViewEvent($element, new TemplateView('template'));
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('template', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_there_is_no_title() : void
+    {
+        $listener = new FrontTitleTeaserListener($this->createFailingConverter());
+
+        $element = $this->loadElement('<front xmlns="http://libero.pub">foo</front>');
+
+        $event = new BuildViewEvent($element, new TemplateView('@LiberoPatterns/teaser.html.twig'));
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('@LiberoPatterns/teaser.html.twig', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_there_is_already_a_teaser_title_set() : void
+    {
+        $listener = new FrontTitleTeaserListener($this->createFailingConverter());
+
+        $element = $this->loadElement('<front xmlns="http://libero.pub"><title>foo</title></front>');
+
+        $event = new BuildViewEvent(
+            $element,
+            new TemplateView('@LiberoPatterns/teaser.html.twig', ['heading' => 'bar'])
+        );
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('@LiberoPatterns/teaser.html.twig', $view->getTemplate());
+        $this->assertSame(['heading' => 'bar'], $view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_sets_the_text_argument() : void
+    {
+        $listener = new FrontTitleTeaserListener($this->createDumpingConverter());
+
+        $element = $this->loadElement(
+            <<<XML
+<libero:front xmlns:libero="http://libero.pub">
```

Comment:
a pity that this is a completely duplicated test in another bundle, that just changes the input from JATS to `libero` XML.

Reply:
Might be able to do something to help these different implementations and their test suites, but definitely one for the future.